.test-infra/tools/stale_bq_datasets_cleaner.sh

```diff
@@ -18,7 +18,7 @@
 #    Deletes stale and old BQ datasets that are left after tests.
 #
 
-set -euo pipefail
```

Comment:
This prints every command executing and I find https://github.com/apache/beam/actions/runs/7731108048/job/21078010193 is verbose and may be difficult to read. We added several "echo" below. Can we keep the original set flags

Reply:
I found that it wasn't enough to diagnose the current issue without debug logging. In general, because we've got -euo set, the script exits on the first non-zero exit code, so seeing the last command was really helpful. I'd vote to keep it